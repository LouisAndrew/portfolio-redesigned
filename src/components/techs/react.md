---
templateKey: techs
title: React
name: ReactJS
icon: assets/react-1-logo-png-transparent.png
---
