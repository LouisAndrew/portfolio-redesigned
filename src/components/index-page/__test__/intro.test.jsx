import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import { ThemeProvider } from 'styled-components'

import { render, cleanup } from '@testing-library/react'
import '@testing-library/jest-dom'

import Intro from '../intro'

describe('Intro / about me element on landing page.', () => {

        afterEach( cleanup )

        const heading = 'About Me'
        const headingList = [
                { h: 'About' },
                { h: 'Me' }
        ]
        const image = 'assets/mock-image.png'

        const par = 'some mock text'
        
        const theme = { center: () => `` }

        const renderWithStyledTheme = ui => <ThemeProvider theme={theme}>{ui}</ThemeProvider> 

        it('renders without crashing', () => {

                const div = document.createElement('div')
                ReactDOM.render(renderWithStyledTheme(<Intro />), div)
        })

        it('renders correctly', () => {

                const { getByTestId, get } = render( renderWithStyledTheme(<Intro heading={heading} headingList={headingList} desc={par} image={image} />) )

                const hdList = getByTestId('intro-heading')

                let i = 0
                hdList.childNodes.forEach( child => {
                        expect(child).toHaveTextContent(headingList[i].h)
                        i++
                } )

                expect( getByTestId('intro-par') ).toHaveTextContent(par)
        })

        it('matches snapshot', () => {

                const tree = renderer.create( renderWithStyledTheme(<Intro heading={heading} headingList={headingList} desc={par} image={image} />) )

                expect(tree).toMatchSnapshot()
        })
})