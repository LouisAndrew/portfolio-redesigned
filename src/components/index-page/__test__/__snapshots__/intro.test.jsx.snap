// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Intro / about me element on landing page. matches snapshot 1`] = `
<section
  className="sc-AxjAm nIveR wrap"
  id="about"
>
  <test-file-stub
    id="corner"
  />
  <div
    className="heading"
    data-testid="intro-heading"
  >
    <h1>
      About
    </h1>
    <h1>
      Me
    </h1>
  </div>
  <div
    className="sc-AxirZ dhjTKK"
    style={
      Object {
        "opacity": 0,
        "transform": "translateY(-100px) translateZ(0)",
      }
    }
  >
    <div
      className="img"
    >
      <div
        className=" gatsby-image-wrapper"
        style={
          Object {
            "maxHeight": null,
            "maxWidth": null,
            "overflow": "hidden",
            "position": "relative",
          }
        }
      >
        <div
          aria-hidden={true}
          style={
            Object {
              "paddingBottom": "NaN%",
              "width": "100%",
            }
          }
        />
        <picture>
          <source />
          <img
            alt=""
            loading="lazy"
            onLoad={[Function]}
            style={
              Object {
                "height": "100%",
                "left": 0,
                "objectFit": "cover",
                "objectPosition": "center",
                "opacity": 0,
                "position": "absolute",
                "top": 0,
                "transition": "opacity 500ms",
                "width": "100%",
              }
            }
          />
        </picture>
        <noscript
          dangerouslySetInnerHTML={
            Object {
              "__html": "<picture><source srcset=\\"undefined\\" /><img loading=\\"lazy\\" src=\\"\\" alt=\\"\\" style=\\"position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
            }
          }
        />
      </div>
    </div>
    <p
      data-testid="intro-par"
    >
      some mock text
    </p>
  </div>
</section>
`;
